wazo_auth/services/all_users.py

```diff
@@ -30,12 +30,19 @@ def update_policies(self, tenant_uuids):
         current_group_policy_associations = (
             self._dao.group.get_all_policy_associations()
         )
+
+        all_users_groups = self._dao.group.get_all_users_groups()
+        all_users_groups_by_tenant = {
+            group.tenant_uuid: group for group in all_users_groups
+        }
+
         for tenant_uuid in tenant_uuids:
             self.associate_policies_for_tenant(
                 tenant_uuid,
                 policies,
                 current_group_policy_associations,
                 existing_config_managed_policies_by_tenant,
+                all_users_groups_by_tenant,
```

Comment:
I believe it would be better to only pass the list of users groups for the current tenant instead of the entire dict. That way `associate_policies_for_tenant()` only have to have the argument `all_users_group`
```suggestion
                all_users_groups_by_tenant[tenant_uuid],
```

Reply:
Agreed, thank you :+1: